README.md

```diff
@@ -1,2 +1,314 @@
+  - There is an issue with executing the SQL query or scanning the result.
+
+### Functionality:
+
+- The `FindLast` function queries the specified database table to retrieve the last record based on the primary key field of the provided structure.
+
+- It checks if the `structure` argument is a pointer to a struct. If not, it returns an error.
+
+- It identifies the primary key field in the provided struct using the "primarykey" tag.
+
+- It constructs an SQL query to retrieve the last record from the specified `tableName` based on the primary key.
+
+- The query results are scanned into the provided `structure`, which should be a pointer to a struct.
+
+### Example Usage:
+
+```go
+var result MyStruct
+err := FindLast("your_table", &result)
+if err != nil {
+    // Handle the error
+}
+```
+## Find Function
+
+The `Find` function queries a database table based on the provided criteria and scans the result into a slice of structs.
+
+### Parameters:
+
+- `data` (`map[string]interface{}`): A map containing the query criteria.
+  - `tablename` (string, required): The name of the database table to query.
+  - `columnname` (string, required): The name of the column to filter on.
+  - `columnvalue` (interface{}, required): The value to filter the `columnname` by.
+  - `sortcolumn` (string, optional): The name of the column to sort the results by. Defaults to the primary key of the table defined in the structure.
+  - `sortvalue` (string, optional): The sorting order, which can be "ASC" (ascending) or "DESC" (descending). Defaults to "ASC".
+
+### Return Value:
+
+- `error`: An error indicating success or failure. Returns `nil` on success.
+
+### Functionality:
+
+- It checks if the required keys (`tablename`, `columnname`, `columnvalue`) are present in the `data` map. If any of these keys are missing, it returns an error.
+
+- It validates that the `structure` argument is a pointer to a slice of structs. If the validation fails, it returns an error.
+
+- If the `sortcolumn` key is not provided in the `data` map, it automatically determines the `sortcolumn` based on the primary key of the table defined in the structure.
+
+- If the `sortvalue` key is not provided in the `data` map, it defaults to "ASC" (ascending) for sorting.
+
+- It constructs a SQL query based on the provided criteria and executes the query on the database.
+
+- The query retrieves rows from the specified `tablename` where the value in the `columnname` matches `columnvalue`. The results are sorted by `sortcolumn` in the specified order (`sortvalue`).
+
+- The query results are scanned into the provided `structure`, which should be a pointer to a slice of structs.
+
+### Example Usage:
+
+```go
+data := map[string]interface{}{
+    "tablename":   "your_table",
+    "columnname":  "name",
+    "columnvalue": "John",
+    "sortcolumn":  "age",
+    "sortvalue":   "DESC",
+}
+
+var result []YourStruct
+err := Find(data, &result)
```

Comment:
This is confusing, how does this function determine which table needs to be run on ?

Reply:
data is type of map[string]interface
data[tablename]
data[columnname]
data[columnvalue]
this keys were need to provided in this so that find will work accordingly